GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTFLLUploader.m

```diff
@@ -328,4 +353,23 @@ - (void)appWillTerminate:(GDTCORApplication *)application {
   });
 }
 
+#pragma mark - NSURLSessionDelegate
+
+- (void)URLSession:(NSURLSession *)session
+                          task:(NSURLSessionTask *)task
+    willPerformHTTPRedirection:(NSHTTPURLResponse *)response
+                    newRequest:(NSURLRequest *)request
+             completionHandler:(void (^)(NSURLRequest *_Nullable))completionHandler {
+  if (!completionHandler) {
+    return;
+  }
+  if (response.statusCode == 302 || response.statusCode == 301) {
+    NSURLRequest *newRequest = [self constructRequestWithURL:request.URL
+                                                        data:task.originalRequest.HTTPBody];
```

Comment:
while testing redirects yesterday I've seen different cases depending on the code. When the code was 307 there might be no `HTTPBody` in either `task.originalRequest`, `task.currentRequest` or `request`, but there was `request.HTTPBodyStream`. When the code was 302 I don't remember either `HTTPBody` or `HTTPBodyStream` for any of the requests instances. Could you double check this code is actually working?

Reply:
I see we keep track of `self.currentTask` meaning we know for sure which task is being processed. Can we store the body as well so we can safely reconstruct the request?

Reply:
The way I instantiate the request (hopefully) keeps the HTTPBody property populated. I can't finish testing until the server begins returning 302s instead of 307s. I'd rather keep the body associated with the request to keep correct context if more than one task ever ends up being executed.

Reply:
Ah, I see, you didn't set the request HTTPBody filed itself. I didn't  notice it before. Can we create task with [this method](https://developer.apple.com/documentation/foundation/nsurlsession/1407613-datataskwithrequest?language=objc) then to remove ambiguity?